src/main/java/io/github/thebusybiscuit/slimefun4/core/config/SlimefunConfigManager.java

```diff
@@ -0,0 +1,275 @@
+package io.github.thebusybiscuit.slimefun4.core.config;
+
+import java.util.function.Supplier;
+import java.util.logging.Level;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.apache.commons.lang.Validate;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Server;
+import org.bukkit.plugin.Plugin;
+
+import io.github.thebusybiscuit.cscorelib2.config.Config;
+import io.github.thebusybiscuit.slimefun4.api.items.ItemSetting;
+import io.github.thebusybiscuit.slimefun4.core.researching.Research;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+
+/**
+ * This class manages our {@link Config}, it caches values for faster access
+ * and provides means to reload it.
+ * 
+ * @author TheBusyBiscuit
+ *
+ */
+public final class SlimefunConfigManager {
+
+    /**
+     * Our {@link SlimefunPlugin} instance
+     */
+    private final SlimefunPlugin plugin;
+
+    /**
+     * Our {@link Plugin} {@link Config} (config.yml)
+     */
+    private final Config pluginConfig;
+
+    /**
+     * Our {@link SlimefunItem} {@link Config} (Items.yml)
+     */
+    private final Config itemsConfig;
+
+    /**
+     * Our {@link Research} {@link Config} (Researches.yml)
+     */
+    private final Config researchesConfig;
+
+    /**
+     * This flag marks whether we require a {@link Server} restart to
+     * apply the latest changes.
+     */
+    private boolean isRestartRequired = false;
+
+    // Various booleans we want to cache instead of re-parsing everytime
+    private boolean isBackwardsCompatibilityEnabled;
+    private boolean isResearchingEnabled;
+    private boolean isResearchingFreeInCreativeMode;
+    private boolean isResearchFireworkEnabled;
+    private boolean isDuplicateBlockLoggingEnabled;
+    private boolean isVanillaRecipeShowingEnabled;
+    private boolean isSlimefunGuideGivenOnJoin;
+    private boolean isUpdaterEnabled;
+    private boolean isTalismanMessageInActionbar;
+    private boolean isExcessCommandItemsDroppingEnabled;
```

Comment:
these shouldn't be named `isX`, I assume this was just a mistake and is from copying methods or whatever over